homework_3/barbhaiwala/solution.js

```diff
@@ -0,0 +1,83 @@
+function NaiveCalculator(equation)
+{
+    this.splitTokens = function()
+    {
+        return equation.split(' ');
+    }
+    
+    this.applyOperation = function (operand1, operation, operand2) {
+        var resultOfOperation;
+        switch(operation)
+        {
+            case '+':
+                resultOfOperation = operand1 + operand2;
+                break;
+            case '-':
+                resultOfOperation = operand1 - operand2;
+                break;
+            case '/':
+                resultOfOperation = operand1 / operand2;
+                break;
+            case 'x':
+            case 'X':
+                resultOfOperation = operand1 * operand2;
+                break;
+            case '^':
+                resultOfOperation = Math.pow(operand1, operand2);
+                break;
+        }
+        
+        return resultOfOperation;
```

Comment:
You could wrap your returns into the switch. Then you wouldn't need to declare and assign to resultOfOperation. Example:
```js
case '+':
   return operand1 + operand2;
```